h/views/activity.py

```diff
@@ -42,9 +42,81 @@ def search(request):
     # Fetch results
     result = query.execute(request, q, page_size=page_size)
 
+    for user in result.aggregations.get('users', []):
+        user['username'] = util.user.split_user(user['user'])['username']
+        del user['user']
+
     return {
         'total': result.total,
         'aggregations': result.aggregations,
         'timeframes': result.timeframes,
         'page': paginate(request, result.total, page_size=page_size),
     }
+
+
+@view_config(route_name='activity.group_search',
+             request_method='GET',
+             renderer='h:templates/activity/search.html.jinja2')
+def group_search(request):
+    if not request.feature('search_page'):
+        raise httpexceptions.HTTPNotFound()
+
+    result = search(request)
+
+    pubid = request.matchdict['pubid']
+
+    try:
+        group = request.db.query(models.Group).filter_by(pubid=pubid).one()
+    except exc.NoResultFound:
+        return result
+
+    if request.authenticated_user not in group.members:
+        return result
+
+    result['group'] = {
+        'created': group.created.strftime('%B, %Y'),
+        'description': group.description,
+        'name': group.name,
+        'pubid': group.pubid,
+    }
+
+    result['show_group_leave_button'] = (
+        group.creator != request.authenticated_user)
+
+    if request.has_permission('admin', group):
+        result['group_edit_url'] = request.route_url('group_edit', pubid=pubid)
+
+    return result
+
+
+@view_config(route_name='activity.group_search',
+             request_method='POST',
+             renderer='h:templates/activity/search.html.jinja2',
+             request_param='group_leave')
```

Comment:
This means that this view function will be called if the browsers POSTs to the "activity.group_search URL (`/groups/<pubid>/search`) _and_ the post includes a param called "group_leave". GET requests to the same URL go to the `group_search()` function above. In future we will add more POST requests with different params instead of "group_leave", for example for adding user or tag facets into the search, and new functions like this one will be added to handle those.